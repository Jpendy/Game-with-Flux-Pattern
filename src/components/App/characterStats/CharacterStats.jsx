import React from 'react';
import PropTypes from 'prop-types';

export default function CharacterStats({ health, attack, experience }) {
  return (
    <div>
      <dl>
        <dt>Health</dt>
        <dd>{health}</dd>
        <dt>Attack</dt>
        <dd>{attack}</dd>
        <dt>Experience</dt>
        <dd>{experience}</dd>
      </dl>
    </div>
  );
}

CharacterStats.propTypes = {
  health: PropTypes.number.isRequired,
  attack: PropTypes.number.isRequired,
  experience: PropTypes.number.isRequired
};
