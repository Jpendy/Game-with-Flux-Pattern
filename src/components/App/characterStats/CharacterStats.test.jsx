import React from 'react';
import { shallow } from 'enzyme';
import CharacterStats from './CharacterStats';

describe('CharacterStats component', () => {
  it('renders CharacterStats', () => {
    const wrapper = shallow(<CharacterStats health={100} attack={10} experience={0} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
