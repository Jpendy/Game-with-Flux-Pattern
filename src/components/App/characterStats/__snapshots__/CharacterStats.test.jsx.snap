// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterStats component renders CharacterStats 1`] = `
<div>
  <dl>
    <dt>
      Health
    </dt>
    <dd>
      100
    </dd>
    <dt>
      Attack
    </dt>
    <dd>
      10
    </dd>
    <dt>
      Experience
    </dt>
    <dd>
      0
    </dd>
  </dl>
</div>
`;
